ReportApp.Web/Controllers/RoleController.cs

```diff
@@ -0,0 +1,216 @@
+        private ApplicationRoleManager _roleManager;
+
+        public RoleController() { }
+        public RoleController(ApplicationUserManager userManager, ApplicationRoleManager roleManager)
+        {
+            UserManager = userManager;
+            RoleManager = roleManager;
+        }
+
+        public ApplicationUserManager UserManager
+        {
+            get
+            {
+                return _userManager ?? HttpContext.GetOwinContext().GetUserManager<ApplicationUserManager>();
+            }
+            set
+            {
+                _userManager = value;
+            }
+        }
+
+        public ApplicationRoleManager RoleManager
+        {
+            get
+            {
+                return _roleManager ?? HttpContext.GetOwinContext().GetUserManager<ApplicationRoleManager>();
+            }
+            set
+            {
+                _roleManager = value;
+            }
+        }
+
+        // GET: Role
+        public ActionResult Index()
+        {
+            return View(RoleManager.Roles);
+        }
+
+        //GET: List of User to a particular role
+        // GET: /Roles/Details/5
+        public async Task<ActionResult> Details(string id)
+        {
+            if (id == null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
+            }
+            var role = await RoleManager.FindByIdAsync(id);
+
+            // Get the list of Users in this Role
+            var users = new List<Profile>();
+
+            // Get the list of Users in this Role
+            foreach (var user in UserManager.Users.ToList())
+            {
+                if (await UserManager.IsInRoleAsync(user.Id, role.Name))
+                {
+                    users.Add(user.Profile);
+                }
+            }
+            ViewBag.Users = users;
+            ViewBag.UserCount = users.Count();
+            return View(role);
+        }
+
+        // GET: /Roles/Create
+        public ActionResult Create()
+        {
+            return View();
+        }
+
+        //
+        // POST: /Roles/Create
+        [HttpPost]
+        public async Task<ActionResult> Create(RoleViewModel roleViewModel)
+        {
+            if (ModelState.IsValid)
+            {
+                //Create Role Name if it does not exist
+                var role = await RoleManager.FindByNameAsync(roleViewModel.Name);
+                if (role == null)
+                {
+                    var newRole = new IdentityRole(roleViewModel.Name);
+                    var result = await RoleManager.CreateAsync(newRole);
+                    if (!result.Succeeded)
+                    {
+                        ModelState.AddModelError("", result.Errors.First());
+                        return View();
+                    }
+                    return RedirectToAction("Index");
+                }
+                ModelState.AddModelError("", "Role Name already Exist!!!");
```

Comment:
1. exist should be exists

2. Convert that into a constant or readonly string. Read about magic numbers on wikipedia or something similar.